torchvision/transforms/functional.py

```diff
@@ -1404,6 +1404,55 @@ def gaussian_blur(img: Tensor, kernel_size: List[int], sigma: Optional[List[floa
     return output
 
 
+def gaussian_noise(img: Tensor, mean: float, sigma: Optional[List[float]]) -> Tensor:
+    """Performs Gaussian blurring on the image by given kernel.
+    If the image is torch Tensor, it is expected
+    to have [..., H, W] shape, where ... means an arbitrary number of leading dimensions.
+
+    Args:
+        img (PIL Image or Tensor): Image to be blurred
+        mean (float): Mean of the desired noise corruption.
+        sigma (sequence of floats or float, optional): Gaussian noise standard deviation. Can be a
+            sequence of floats like ``(sigma_x, sigma_y)`` or a single float to define the
+            same sigma in both X/Y directions.
+
+            .. note::
+                In torchscript mode sigma as single float is
+                not supported, use a sequence of length 1: ``[sigma, ]``.
+
+    Returns:
+        PIL Image or Tensor: Gaussian Blurred version of the image.
+    """
+    if not torch.jit.is_scripting() and not torch.jit.is_tracing():
+        _log_api_usage_once(gaussian_noise)
+
+    if sigma is None:
+        raise ValueError("The value of sigma cannot be None.")
+
+    if sigma is not None and not isinstance(sigma, (int, float, list, tuple)):
+        raise TypeError(f"sigma should be either float or sequence of floats. Got {type(sigma)}")
```

Comment:
Something is unclear here about whether sigma is `Optional` or not. Docstring and typehint says optional, but you raise an error if sigma is None. Let's remove optional and assume that sigma is not None

Reply:
@parth-shastri check above comment and the code. Why do you check `sigma is None` if sigma is not intended to be `Optional`.